pandas/tests/groupby/test_function.py

```diff
@@ -1015,6 +1025,81 @@ def test_nunique_with_timegrouper():
     tm.assert_series_equal(result, expected)
 
 
+@pytest.mark.parametrize(
```

Comment:
The parametrization here is pretty repetitive, though I realize that you have three items at a time being sent through to keep the expectation different across each.

Is there a way to more succinctly parametrize though? It's rather difficult to read this and find what's expected

Reply:
Agreed - have modified it so the DataFrame is constructed within the test